rust/lance-core/src/utils/mask.rs

```diff
@@ -63,6 +63,23 @@ impl RowIdMask {
         }
     }
 
+    // If there is both a block list and an allow list then collapse into just an allow list
+    pub fn normalize(self) -> Self {
+        if let Self {
+            allow_list: Some(mut allow_list),
+            block_list: Some(block_list),
+        } = self
+        {
+            allow_list -= &block_list;
+            Self {
+                allow_list: Some(allow_list),
+                block_list: None,
+            }
+        } else {
+            self
+        }
+    }
```

Comment:
I suppose this begs the question why we allow there to be both anyways.  It would simplify a few paths if we collapse RowIdMask into an enum of `All, Allow, Block`.

Reply:
does that assume the BlockList is a subset of the AllowList?

Reply:
I guess the two lists are exhaustive for the fragments covered by the index, so we don't have an issue here.